controllers/workspace/devworkspace_controller.go

```diff
@@ -240,7 +242,27 @@ func (r *DevWorkspaceReconciler) Reconcile(req ctrl.Request) (reconcileResult ct
 		return reconcile.Result{Requeue: true}, nil
 	}
 
+	annotate.AddURLAttributesToEndpoints(&workspace.Spec.Template, routingStatus.ExposedEndpoints)
```

Comment:
it seems to have side effects:

<details>

<summary>Create DW with local plugin</summary>

cat EOF >> kubectl apply -f

---
apiVersion: workspace.devfile.io/v1alpha2
kind: DevWorkspace
metadata:
  name: theia
  namespace: user1-che
spec:
  started: true
  template:
    components:
    - name: nodejs
      container:
        image: quay.io/eclipse/che-nodejs10-ubi:nightly
        memoryLimit: 512Mi
        endpoints:
        - name: nodejs
          protocol: http
          targetPort: 3000
        mountSources: true
    - name: theia-ide-workspace05c266f85ef741bd
      plugin:
        kubernetes:
          name: theia-ide-local-plugin
          namespace: user1-che
    commands:
    - id: download-dependencies
      exec:
        component: nodejs
        commandLine: npm install
        workingDir: "${PROJECTS_ROOT}/project/app"
    - id: run-the-app
      exec:
        component: nodejs
        commandLine: nodemon app.js
        workingDir: "${PROJECTS_ROOT}/project/app"
    - id: run-the-app-with-debugging-enabled
      exec:
        component: nodejs
        commandLine: nodemon --inspect app.js
        workingDir: "${PROJECTS_ROOT}/project/app"
    - id: stop-the-app
      exec:
        component: nodejs
        commandLine: 'node_server_pids=$(pgrep -fx ''.*nodemon (--inspect )?app.js''
          | tr "\\n" " ") && echo "Stopping node server with PIDs: ${node_server_pids}"
          && kill -15 ${node_server_pids} &>/dev/null && echo ''Done.'''
---
kind: DevWorkspaceTemplate
apiVersion: workspace.devfile.io/v1alpha2
metadata:
  name: theia-ide-local-plugin
  namespace: user1-che
spec:
  commands:
  - id: init-container-command
    apply:
      component: remote-runtime-injector
  events:
    preStart:
    - init-container-command
  components:
  - name: remote-endpoint
    volume:
      ephemeral: true
  - name: remote-runtime-injector
    container:
      image: quay.io/eclipse/che-theia-endpoint-runtime-binary:next
      env:
      - name: PLUGIN_REMOTE_ENDPOINT_EXECUTABLE
        value: "/remote-endpoint/plugin-remote-endpoint"
      - name: REMOTE_ENDPOINT_VOLUME_NAME
        value: remote-endpoint
      volumeMounts:
      - name: remote-endpoint
        path: "/remote-endpoint"

EOF

</details>
once it's started, check that `kubectl get dw theia` has `remote-runtime-injector` component in it's spec.template.

Reply:
also, IDE url is not propagate to status for some reason, not sure if it's related to this PR. Checking on main branch 
```
theia   workspaced091e53ca11849ad   Running   Workspace is running
```

Reply:
https://github.com/devfile/devworkspace-operator/pull/326

Reply:
should be fixed after you rebase